nltk/tokenize/casual.py

```diff
@@ -262,16 +270,34 @@ def tokenize(self, text):
                               x.lower()), words))
         return words
 
+######################################################################
+# Normalization Functions
+######################################################################
+
+def reduce_lengthening(text):
+    '''
+    Replace character sequences of length 3 or greater with sequences of length 3.
+    '''
+    pattern = re.compile(r"(.)\1{2,}")
+    return pattern.sub(r"\1\1\1", text)
+
+def remove_handles(text):
+    '''
+    Remove twitter username handles from text.
+    '''
+    pattern = re.compile(r"(?<=^|(?<=[^a-zA-Z0-9-\.]))@([A-Za-z_]+[A-Za-z0-9_]+)")
```

Comment:
Nitpicks:
* Don't nest look-behinds: `^` doesn't need a lookbehind, and can just be a normal (non-capturing) group.
* The capturing group for the username is unnecessary.
* Should `_` be included in `[^a-zA-Z0-9-\.]`? If so, this can be written as `[^\w.-]`, just as `[A-Za-z0-9_]` can be written `\w`. I suspect, though, that the lookbehind expression should actually include all unicode word characters, while the username expression should be ASCII exclusively.
* a parameter for substituting with a placeholder (like `'<USER>'`) rather than deleting might make this more useful (and could be allowed as a value for `strip_handles` param rather than True).
* this will result in double-spaces after deletions; do we care? Handling them could be fussy.
* `strip_handles` param and `remove_handles` function should probably have the same name
* again, a doctest example would be nice.